core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -177,9 +178,21 @@ impl GasAdjuster {
         let calculated_price =
             (self.config.internal_l1_pricing_multiplier * effective_gas_price as f64) as u64;
 
-        let conversion_rate = self.base_token_fetcher.conversion_rate().unwrap_or(1);
+        let conversion_rate = self
+            .base_token_fetcher
+            .conversion_rate()
+            .unwrap_or(BigDecimal::from(1));
+        let bound_gas_price = BigDecimal::from(self.bound_gas_price(calculated_price));
 
-        self.bound_gas_price(calculated_price) * conversion_rate
+        U256::from_dec_str(
+            &match (conversion_rate * bound_gas_price).round(0) {
+                zero if zero == BigDecimal::from(0) => BigDecimal::from(1),
+                val => val,
+            }
+            .to_string(),
+        )
+        .unwrap()
```

Comment:
Can't we return a Result<T, Err> instead of unwrapping?
Maybe an error can fall back to the previously set value, we probably have discussed this already, but wdy?

Reply:
Currently we `unwrap()` because we assume that the calculated gas prices won't overflow `u64`, this will only happen if the conversion rate is between what's specified as safe in the documentation.

If we were to return a `Result` then [get_fee_model_params](https://github.com/lambdaclass/zksync-era/blob/fa90d1653ed08a87cc4a9463200167e71930764b/core/node/fee_model/src/lib.rs#L66-L78) would need to handle this `Result`, which might cascade into a number of functions having to handle `Results` when they previously didn't.